File: Csharp/TwoSum.cs
using System.Collections;
using System.Collections.Generic;

namespace Csharp
{
    public class Solution
    {
        public int[] TwoSum(int[] nums, int target)
        {
            Hashtable hashtable = new Hashtable();
            for (int i = 0; i < nums.Length; i++)
            {
                if (hashtable.ContainsKey(target - nums[i]))
                {
                    return new int[] { (int)(hashtable[(target - nums[i])]), i };
                }
                else if(hashtable.ContainsKey(nums[i]))
                {
                    hashtable[nums[i]] = i;
                }
                else
                {
                    hashtable.Add(nums[i], i);
                }
            }

            return null;
        }

        public int[] TwoSum2(int[] nums, int target)
        {
            Dictionary<int,int> dictionary = new Dictionary<int, int>();
            
            for (int i = 0; i < nums.Length; i++)
            {
                if (dictionary.ContainsKey(target - nums[i]))
                {
                    return new int[] { dictionary[(target - nums[i])], i };
                }
                else if(dictionary.ContainsKey(nums[i]))
                {
                    dictionary[nums[i]] = i;
                }
                else
                {
                    dictionary.Add(nums[i], i);
                }
            }

            return null;
        }
    }
}

File: Python3/TwoSum.py
from typing import List

class Twosum:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hm = {}
        for i,v in enumerate(nums):
            n = target-v
            if n in hm:
                return [i,hm[n]]
            else:
                hm[v] = i
        return []
       
            
test = Twosum()
test.twoSum([2,7,11,15],9)   